specification/electronic-prescription-service-api.yaml

```diff
@@ -37,14 +37,25 @@ info:
     ## Digital signatures
     Electronic prescription messages are digitally signed to protect against tampering and to prevent repudiation. While the messages themselves are FHIR-based, for reasons of backwards compatibility, the signatures must be represented in HL7 V3 format. To make this easier, you can POST a prescription to the `/$prepare` endpoint to receive exactly the HL7 V3 fragments which need to be signed to conform to the specification.
 
-    To generate a signature for your prescription, you can use the digital signature service. To access the service, [contact us](https://digital.nhs.uk/developer/help-and-support).
+    To generate a signature for your prescription, you can use the Signing Service. To access the service, [contact us](https://digital.nhs.uk/developer/help-and-support).
 
     ## Using this API
     To create a prescription:
-    1. POST a [prescription-order message](https://simplifier.net/guide/DigitalMedicines/prescription-order) to the `/$prepare` endpoint
-    2. sign the response and put the result in a `Provenance` resource
-    3. add the `Provenance` to the prescription-order message
-    4. POST the signed prescription-order message to the `/$process-message` endpoint
+    1. POST a [prescription-order message](https://simplifier.net/guide/DigitalMedicines/prescription-order) to the `/$prepare` endpoint to get the encoded prescription data that requires a signature
+    2. sign the encoded prescription data using the Signing Service to get the signed prescription data
+    3. generate an XML-DSig using the signed prescription data (see below)
+    4. put the XML-DSig in a `Provenance` resource
+    5. add the `Provenance` to the prescription-order message
+    6. POST the signed prescription-order message to the `/$process-message` endpoint
+
+    Example code to generate an XML-DSig: 
+    ```private fun toXmlSignature(digest: ByteArray, signature: ByteArray, certificate: ByteArray): ByteArray {
```

Comment:
I like the code example :+1:.

Would be good if we told the supplier where to get each of the three parameters. E.g. "digest" comes from EPS `POST /$prepare`, "signature" and "certificate" come from Signing Service `GET SignatureResponse`.

Also, the method could probably take three strings instead of three ByteArrays. The "digest" which comes back from prepare is already a string, and the "signature" and "certificate" which come back from SignatureResponse are already base64 encoded strings.

Reply:
Also we should base64 encode the final result before it goes in the Provenance.signature.data.